src/components/Tabs/AnnotationTab.tsx

```diff
@@ -0,0 +1,360 @@
+    const index = createNewLabel();
+    setCurrentLabelIdx(index);
+    setShowEditPopover(false);
+  };
+
+  const onDeleteLabel = (): void => {
+    if (currentLabelIdx !== null) {
+      deleteLabel(currentLabelIdx);
+    }
+    setShowEditPopover(false);
+  };
+
+  const onClickObjectRow = (_event: React.MouseEvent<any, MouseEvent>, record: TableDataType): void => {
+    props.setTrackAndFrame(record.track, record.time);
+  };
+
+  const onClickEditButton = (): void => {
+    setShowEditPopover(!showEditPopover);
+    setEditPopoverNameInput(selectedLabel?.name || "");
+  };
+
+  useEffect(() => {
+    if (showEditPopover) {
+      editPopoverInputRef.current?.focus();
+    }
+  }, [showEditPopover]);
+
+  const onClickEditCancel = (): void => {
+    setShowEditPopover(false);
+  };
+
+  const onClickEditSave = (): void => {
+    if (currentLabelIdx !== null) {
+      setLabelName(currentLabelIdx, editPopoverNameInput);
+    }
+    setShowEditPopover(false);
+  };
+
+  const tableColumns: TableProps<TableDataType>["columns"] = [
+    {
+      title: "Object ID",
+      dataIndex: "id",
+      key: "id",
+      sorter: (a, b) => a.id - b.id,
+    },
+    {
+      title: "Track ID",
+      dataIndex: "track",
+      key: "track",
+      sorter: (a, b) => a.track - b.track,
+    },
+    {
+      title: "Time",
+      dataIndex: "time",
+      key: "time",
+      sorter: (a, b) => a.time - b.time,
+    },
+    // Column that contains a remove button for the ID.
+    {
+      title: "",
+      key: "action",
+      render: (_, record) => (
+        <IconButton
+          type="text"
+          onClick={(event) => {
+            // Rows have their own behavior on click (jumping to a timestamp),
+            // so we need to stop event propagation so that the row click event
+            // doesn't fire.
+            event.stopPropagation();
+            setLabelOnId(currentLabelIdx!, record.id, false);
+          }}
+        >
+          <CloseOutlined />
+          <VisuallyHidden>
+            Remove ID {record.id} (track {record.track})
+          </VisuallyHidden>
+        </IconButton>
+      ),
+    },
+  ];
+
+  const tableData: TableDataType[] = useMemo(() => {
+    const dataset = props.dataset;
+    if (currentLabelIdx !== null && dataset) {
+      const ids = annotationData.getLabeledIds(currentLabelIdx);
+      return ids.map((id) => {
+        const track = dataset.getTrackId(id);
+        const time = dataset.getTime(id);
+        return { key: id.toString(), id, track, time };
+      });
+    }
+    return [];
+  }, [annotationData, currentLabelIdx, props.dataset]);
+
+  // Options for the selection dropdown
+  const selectLabelOptions: ItemType[] = labels.map((label, index) => {
+    return {
+      key: index.toString(),
+      label: label.ids.size ? `${label.name} (${label.ids.size})` : label.name,
+    };
+  });
+
+  // A small popup that appears when you press the edit button.
+  const editPopoverContents = (
```

Comment:
also could be it's own file

Reply:
Leaving it for now but if more form inputs are needed down the line I can make that change.